plugins/providers/bigquery/model.go

```diff
@@ -32,18 +33,34 @@ func (d *Dataset) FromDomain(r *domain.Resource) error {
 }
 
 func (d *Dataset) ToDomain() *domain.Resource {
-	return &domain.Resource{
+	r := &domain.Resource{
 		Type: ResourceTypeDataset,
 		Name: d.DatasetID,
 		URN:  fmt.Sprintf("%s:%s", d.ProjectID, d.DatasetID),
 	}
+
+	var metadata map[string]interface{}
+	if d.Labels != nil {
+		metadata = map[string]interface{}{
+			"labels": d.Labels,
+		}
+	}
+	if metadata != nil {
```

Comment:
I didn't get this check. At this point, isn't `Metadata` always nil because of line 42? 🤔

Reply:
I think I get it. But it feels confusing to check this way. Can this be done differently ?
Something like - 

```go
	var metadata map[string]interface{}
	if d.Labels != nil {
		metadata = map[string]interface{}{
			"labels": d.Labels,
		}
		if r.Details == nil {
			r.Details = make(map[string]interface{})
		}
		r.Details["_metadata"] = metadata
	}

```

Reply:
@bsushmith updated, keeping it simple for now 👍